### ISSUE TITLE
Trudesk API ISSUE Create User Not Working

### ISSUE THREAD

**Issue by gaurav0914** [polonel/trudesk#120]
**Is this a BUG REPORT or FEATURE REQUEST?**:
No

- [x] BUG 
- [ ] FEATURE

**What happened**:
I am working on Trudesk application integration for User provisioning using trudesk rest API ,while testing the Create user API functionality ,I found that in request data i am trying to set value of "aGrps" as given in API document as follows 

{     "aUsername":    "test.user",     "aPass":        "password",     "aPassConfirm": "password",     "aFullname":    "testuser",     "aEmail":       "testuser@email.com",     "aRole":        "{Support}",     "aTitle":       "support user",     "aGrps":        "[{5babc088867e5008141172b3}]"}
it is unable to set Group ID and i am getting 400 type error ,I also tried different combination but still the groups are not getting set .please suggest an example what format should i pass in "aGrps"

**What did you expect to happen**:

In request data "aGrps" value is not getting set due to which API throughs 400 bad request error.

**How to reproduce it (as minimally and precisely as possible)**:
please try to pass the below data in Soap API tool and test if the user is getting created and the group are being set or not. Role/Group id value should be as per your environment

{   
  "aUsername":    "test.user",     "aPass":        "password",     "aPassConfirm": "password",     "aFullname":    "testuser",     "aEmail":       "testuser@email.com",     "aRole":        "{Support}",     "aTitle":       "support user",     "aGrps":        "["{5babc088867e5008141172b3}]"
[Trudesk API Issue Case.docx](https://github.com/polonel/trudesk/files/2667374/Trudesk.API.Issue.Case.docx)
 
}

**Anything else we need to know?**:
NA
**Environment**:test and development
- Trudesk Version:1.10 Virtual Appliance
- OS (e.g. from /etc/os-release):Alpine linux3.7 kernel 4.9.73-0-virthardend
- Node.JS Version: 4.2.0
- MongoDB Version:3.6.3
- Is this hosted on cloud.trudesk.io: No

**Comment by polonel** [comment]
Sorry, this is a confusing aspect of the documentation that is long overdue for an update.

Where ever I used `{SomethingID}` I put it inside `{  }` and that was a mistake on my part. It was meant as a placeholder for when an ObjectID was needed. Just remove the `{` and `}` from your request where ever an ID is being set.

**Comment by gaurav0914** [comment]
Hi Chris,

I have remmoved { and } from my request,I have now passed agrps value in request data as follows : "aGrps":["5babc088867e5008141172b3"] in this case user is getting created but unable to set user group please help what should be passed in req data to set user group.

refer to the screenshot of request data and response attached.
![trudeskcreate user](https://user-images.githubusercontent.com/40995164/49850012-001b9f00-fe02-11e8-86ea-e3a8339ede20.PNG)

**Comment by gaurav0914** [comment]
Hi Chris,

Please ignore my previous comment screenshot please find the new one in this comment section

I have remmoved { and } from my request,I have now passed agrps value in request data as follows : "aGrps":["5babc088867e5008141172b3"] in this case user is getting created but unable to set user group please help what should be passed in req data to set user group.

refer to the screenshot of request data and response attached.
![trudeskcreate user](https://user-images.githubusercontent.com/40995164/49850191-96e85b80-fe02-11e8-9b51-e0f6a62bc1dd.PNG)
![userkg16](https://user-images.githubusercontent.com/40995164/49850204-a4054a80-fe02-11e8-9511-f2ee2932d96b.PNG)

**Comment by polonel** [comment]
I have tested this with the latest version (which nothing has changed since 0.1.10).

I think you have the incorrect _id in the group array. Double check to make sure you didn't pull a member ID and actually pulled the group id. I've made this mistake before, so I tend to collapse the member array making it easier to see the group _id.

![groupid](https://user-images.githubusercontent.com/626303/49892719-a45c1080-fe17-11e8-9c15-c91c3e950b19.JPG)

![createuser](https://user-images.githubusercontent.com/626303/49892723-a7570100-fe17-11e8-99f9-8f234fc9893d.JPG)
